site-src/geps/gep-1282.md

```diff
@@ -98,12 +106,292 @@ Of course, from the section above, you can see that there are many other feature
 
 ## API
 
-To be written later, once we have agreement on the “what” and the “why”. (This is the “how”).
-
+A new object, BackendCapabilities, is added to the API that serves as a wrapper to a Kubernetes Service, and provisions
+backend capabilities for a Service that requires them.  As discussed by the Gateway API maintainers and community, this
+architectural choice is made because we would ideally like to locate this capabilities set within the Service but until
+that is possible in Kubernetes, we can start with a wrapper to the Service.
+
+BackendCapabilities contains a BackendCapabilitiesSpec object that specifies the targeted backend and defines each of
+the different backend capabilities.  In this GEP, TLS encryption between gateway and backend is proposed as a capability, 
+but with a design that allows additional capabilities to be added as needed in subsequent versions of
+Gateway API.  BackendCapabilities also contains a BackendCapabilitiesStatus which holds the current state.
+
+In https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1beta1/shared_types.go add a new type:
+```go
+// BackendCapabilities references a Kubernetes Service and defines any backend 
+// capabilities that Service requires.  
+//
+// +optional
+type BackendCapabilities struct {
+    // Name is the name of this backend capability set.
+    // +required
+    Name string `json:”name”` 
+	
+    // Spec specifies the capabilities supported by this BackendCapabilities
+    // Service wrapper, if any.  The capabilities are predefined, and have two
+    // characteristics in common: they are tightly bound to the backend, 
+    // rather than being important at the route level, and the service owner
+    // (application developer in the Gateway API personas) should control 
+    // these settings, not the owner of the Gateway implementation (the 
+    // Cluster Operator in the Gateway API personas).
+    // 
+    // Support: Extended
+    // 
+    // +required
+    Spec BackendCapabilitiesSpec `json:”spec”`
+
+    // Status represents the current state of the backend and its capabilities.
+    // +optional
+    Status *BackendCapabilitiesStatus `json:”status,omitempty”`
+}
+
+// BackendTargetReference identifies an API object to apply BackendCapabilities to. 
+// 
+type BackendTargetReference struct {
+    // Group is the group of the target resource.
+    Group Group `json:"group"`
+
+    // Kind is kind of the target resource.
+    // Valid values include:
+    // * "Service" (the default if no Kind is specified)
```

Comment:
How can it be a Service if it's a backend/workload capability ? Deployment/ReplicaSet/etc - or pod label selector maybe ? 

If you want this to be a Service extension - call it ServiceProperties or ServiceTLS, etc. 

At least in Istio we make a very big distinction between workloads - pods selected by label - and Services.  I think  many users treat Service as a frontend - and the selected Pods as backend.

Reply:
This is absolutely the Service being used as a backend, as a shortcut for a selector. That's why we called it _Backend_Capabilities, not just _Service_Capabilities. Remember, a Service is both the _frontend_ (what you spend most of your time thinking of it as, because that's what Istio needs), and the _backend_, a bucket of workload endpoints (which is how people who aren't doing service mesh tend to see it more).

Having this reference a Service resource is a way avoid having to store the workload selector in two places, as well as a way to meet people who are using Ingress controllers for north-south traffic more than they are Services for east-west traffic.